src/main/java/org/kohsuke/github/GHCommitIterable.java

```diff
@@ -0,0 +1,72 @@
+package org.kohsuke.github;
+
+import java.util.Iterator;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Iterable for commit listing.
+ *
+ * @author Stephen Horgan
+ */
+class GHCommitIterable extends PagedIterable<GHCommit.File> {
+
+    private final GHRepository owner;
+    private final String sha;
+    private GHCommitFilesPage result;
+
+    /**
+     * Instantiates a new GH commit iterable.
+     *
+     * @param owner
+     *            the owner
+     * @param sha
+     *            the SHA of the commit
+     */
+    public GHCommitIterable(GHRepository owner, String sha) {
+        this.owner = owner;
+        this.sha = sha;
+    }
```

Comment:
Need to add an internal constructor for this path.
```suggestion
    public GHCommitIterable(GHRepository owner, String sha, List<File> files) {
        this.owner = owner;
        this.sha = sha;
        this.result = new GHCommitFilesPage(files != null ? files.toArray(GHCommit.File[]::new) : new GHCommit.File[0]);
    }
```